sklearn/metrics/tests/test_regression.py

```diff
@@ -583,19 +585,20 @@ def objective_func(x):
         return mean_pinball_loss(data, constant_pred, alpha=target_quantile)
 
     result = optimize.minimize(objective_func, data.mean(), method="Nelder-Mead")
+    result_x = result.x
     assert result.success
     # The minimum is not unique with limited data, hence the large tolerance.
-    assert result.x == pytest.approx(best_pred, rel=1e-2)
+    assert result_x == pytest.approx(best_pred, rel=1e-2)
```

Comment:
I think the issue is mostly with recovering the actual value which can turn out to be somewhat different than the actual best_pred. Moreover the optimization might also not be converging to the actual minimum loss on some occasions using the default tolerance in the minimize method. 
What would be the best approach to handling this @ogrisel @lorentzenchr

Reply:
Given the approximate nature of this part of the test due to the limited data, increasing the tol looks fine to me. I also added an absolute tol as well to take into account the normal distribution + 0.5 quantile (i.e. expected close to 0).